routers/web/org/setting.go

```diff
@@ -251,3 +230,26 @@ func SettingsRenamePost(ctx *context.Context) {
 	ctx.Flash.Success(ctx.Tr("org.settings.rename_success", oldOrgName, newOrgName))
 	ctx.JSONRedirect(setting.AppSubURL + "/org/" + url.PathEscape(newOrgName) + "/settings")
 }
+
+// SettingsChangeVisibilityPost response for change organization visibility
+func SettingsChangeVisibilityPost(ctx *context.Context) {
+	visibility := structs.VisibilityModes[ctx.FormString("visibility")]
+	if !visibility.IsValid() {
+		ctx.JSONError(ctx.Tr("org.settings.invalid_visibility"))
+		return
+	}
+
+	if ctx.Org.Organization.Visibility == visibility {
+		ctx.JSONError(ctx.Tr("org.settings.change_visibility_no_change"))
```

Comment:
It shouldn't be an "error". It can just flash a info or success message.

And there can be a general message that "You did not make any changes"